test/core/iomgr/mpmcqueue_test.cc

```diff
@@ -119,6 +119,50 @@ static void test_FIFO(void) {
   }
 }
 
+static void test_space_efficiency(void) {
+  gpr_log(GPR_INFO, "test_space_efficiency");
+  grpc_core::InfLenFIFOQueue queue;
+  for (int i = 0; i < 1024; ++i) {
+    queue.Put(static_cast<void*>(grpc_core::New<WorkItem>(i)));
+  }
+  GPR_ASSERT(queue.num_node() == 1024);
+  for (int i = 0; i < 1024; ++i) {
+    WorkItem* item = static_cast<WorkItem*>(queue.Get());
+    queue.Put(item);
+  }
+  GPR_ASSERT(queue.num_node() == 1024);
```

Comment:
Please add more comments about your expectations in this test. Why num_node() should still be 1024 here? This will help future owners to understand your test cases.

Also, 1024 is kind of a magic number here. If we change the implementation of mpmcqueue and use a different number there, this test case needs to be updated too. Please document why you choose these numbers (1024, 4000, ...) here.